### ISSUE TITLE
GC's own internal clock time speeds up (either bios or Nintendont makes it happen)

### ISSUE THREAD

**Issue by Mimimax** [FIX94/Nintendont#831]
When playing a game that uses the internal clock such as Animal Crossing, the clock ocasionally goes over up itself a little in time. Now, that could be either the Animal Crossing game itself or due to the use of the IPL bin (US in my case), I'm not entirely sure. My Wii's own clock either doesn't go up in time, or does it much less often. Also I use a real 1st party 251 block memory card, if it matters. I always use a 1st party GC controller, but sometimes my wiimote is on due to entering the game while having it on sometimes, wheter the Classic Controller is connected or not, in case it could influence the internal clock's time speed.

**Comment by MarsBars06** [comment]
Hi @Mimimax
Yeah, I get this as well.
The original GameCube runs at 485Mhz and the Wii runs at 729Mhz
Maybe the bios is running using the Wii clock speed instead of the actual GameCube clock speed?

**Comment by Mimimax** [comment]
Maybe. I'll see if not using the IPL corrects it.

**Comment by Mimimax** [comment]
Maybe. I could test it without the IPL to see if it is, though that'll take a while to notice in the game.

**Comment by Mimimax** [comment]
Yeah, without the GC IPL on, the game's clock does then follow the Wii's battery clock instead, without the big advance. Now, I just can't be sure if the Wii's own clock going faster slighly (but much slower than the IPL's) is happening due do Nintendont, USB Loader GX, or if it's just my Wii's battery itself. 

If anything can be done to Ninendont to help with either or both the Wii's clock and the IPL's on this case, I'd much appreciate it. I really like to be able use the GC IPL, but this issue has become such a bother when playing Animal Crossing that I had no choice but to disable it for the time being.

**Comment by MarsBars06** [comment]
Hi @Mimimax 
If it follows normal time without booting through the IPL bios, it could just be the IPL bios dump.
Is there any way to verify that your IPL bios is a proper dump?
I'm using wiiflow lite and I get this problem so it most likely isn't USBLoaderGX playing up.

**Comment by Mimimax** [comment]
Hi @MarsBars06
Yes, it's a clean US IPL dump. 
Now, it's one of the older versions I've been using, with the pink cubes. I can't say for sure if that could have to do with the clock desync from the Wii's clock while using it via Nintendont. Looking on the gc forever wiki page (for example) about them doesn't really mention glitch fixes on later versions, just technical specifications. So this would require more testing with them if it has to do with the IPL. I can test other versions if it's needed.
So if loaders don't have any influence that must mean it's the IPL and/or Nintendont itself.

**Comment by Mimimax** [comment]
Ok so, with the latest US IPL rev in use, I corrected the clock in it before starting the game cause it was a bit ahead, and so far it just sped up less than a minute, just like the clock in my Wii. So either it's cause I changed the clock to the correct time beforehand (even though it wasnt needed with the first rev), or using the later rev fixes the issue, or both. It's been about two weeks and it's ok still. So I'm guessing it does have to do with the rev of the IPL being used then.
